crates/core/src/operations/optimize.rs

```diff
@@ -740,24 +792,44 @@ impl MergePlan {
                 }
 
                 table.update().await?;
-                debug!("committing {} actions", actions.len());
-                //// TODO: Check for remove actions on optimized partitions. If a
-                //// optimized partition was updated then abort the commit. Requires (#593).
-                commit(
-                    table.log_store.as_ref(),
-                    &actions,
-                    self.task_parameters.input_parameters.clone().into(),
-                    Some(table.snapshot()?),
-                    Some(app_metadata.clone()),
-                )
-                .await?;
+
+                if commit_writes {
+
+                    debug!("committing {} actions", actions.len());
+                    //// TODO: Check for remove actions on optimized partitions. If a
+                    //// optimized partition was updated then abort the commit. Requires (#593).
+                    commit(
+                        table.log_store.as_ref(),
+                        &actions,
+                        self.task_parameters.input_parameters.clone().into(),
+                        Some(table.snapshot()?),
+                        Some(app_metadata.clone()),
+                    )
+                    .await?;
+
+                } else {
+                    // Save the actions buffer so the client can commit later
+                    total_actions.extend(actions)
+                }
+
             }
 
             if end {
                 break;
             }
         }
 
+        let commit_context = if !commit_writes {
+                Some(CommitContext{
+                        actions: total_actions,
+                        app_metadata,
+                        snapshot: Some(snapshot.clone()), // using original table snapshot as all commits are added at once
```

Comment:
Not sure if I should be used `table.snapshot()?` instead of `snapshot.clone()`. 

`table.snapshot()` is the updated version, if any commits happened while this was running, they won't be picked up by `snapshot.clone()`, is that correct?